### PR TITLE
Add typescript definitions

### LINKED ISSUE #3: Add Typescript support
Types are fun

### PR DESCRIPTION
**Pull request by markwylde**
I'm not sure if all the types are correct. I have used `boolean` where there was a default `true` or `false` given, then `string` for the rest. Let me know if I've got any wrong and I'll fix.

Resolves https://github.com/votemike/lh3-googleusercontent-url-builder/issues/3

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  height: string,
+  flipHorizontally: boolean,
+  flipVertically: boolean,
+  forceDownload: boolean,
+  forceLossyCompression: boolean,
+  forceScaling: boolean,
+  imageFormat: string,
+  losslessCompression: string,
+  lossyCompressionLevel: string,
+  lossyCompressionVersion: string,
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  lossyCompressionVersion: 0 | 1 | 2 | 3,
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  disableWebP: boolean,
+  height: string,
+  flipHorizontally: boolean,
+  flipVertically: boolean,
+  forceDownload: boolean,
+  forceLossyCompression: boolean,
+  forceScaling: boolean,
+  imageFormat: string,
+  losslessCompression: string,
+  lossyCompressionLevel: string,
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  lossyCompressionLevel: number,
```
A percentage between 0 and 100

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  lossyCompressionVersion: string,
+  padding: string,
+  paddingColor: string,
+  rotate: string,
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  rotate: 90 | 180 | 270,
```

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  square: string,
+  upscaling: boolean,
+  width: string
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  width: number
```

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  forceLossyCompression: boolean,
+  forceScaling: boolean,
+  imageFormat: string,
+  losslessCompression: string,
+  lossyCompressionLevel: string,
+  lossyCompressionVersion: string,
+  padding: string,
+  paddingColor: string,
+  rotate: string,
+  square: string,
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  square: number,
```

File: index.d.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  flipHorizontally: boolean,
+  flipVertically: boolean,
+  forceDownload: boolean,
+  forceLossyCompression: boolean,
+  forceScaling: boolean,
+  imageFormat: string,
+  losslessCompression: string,
+  lossyCompressionLevel: string,
+  lossyCompressionVersion: string,
+  padding: string,
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  padding: number,
```

File: index.d.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  paddingColor: string,
```
Comment (votemike) [inline index.d.ts:22]: This will be a hex value

File: index.d.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  cropWithDifferentFocus: boolean,
+  disableWebP: boolean,
+  height: string,
+  flipHorizontally: boolean,
+  flipVertically: boolean,
+  forceDownload: boolean,
+  forceLossyCompression: boolean,
+  forceScaling: boolean,
+  imageFormat: string,
+  losslessCompression: string,
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  losslessCompression: boolean,
```

File: index.d.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  crop: string,
+  cropWithDifferentFocus: boolean,
+  disableWebP: boolean,
+  height: string,
+  flipHorizontally: boolean,
+  flipVertically: boolean,
+  forceDownload: boolean,
+  forceLossyCompression: boolean,
+  forceScaling: boolean,
+  imageFormat: string,
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  imageFormat: imageFormats,
```

File: index.d.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+  allowUpscale: boolean,
+  backgroundColor: string,
+  borderRadius: string,
+  borderWidth: string,
+  circularMask: string,
+  color: string,
+  crop: string,
+  cropWithDifferentFocus: boolean,
+  disableWebP: boolean,
+  height: string,
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  height: number,
```

File: index.d.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
+interface UrlBuilderOptions {
+  allowUpscale: boolean,
+  backgroundColor: string,
+  borderRadius: string,
+  borderWidth: string,
+  circularMask: string,
+  color: string,
+  crop: string,
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  crop: boolean,
```

File: index.d.ts [hunk 12]
```diff
@@ -0,0 +1,29 @@
+interface UrlBuilderOptions {
+  allowUpscale: boolean,
+  backgroundColor: string,
+  borderRadius: string,
+  borderWidth: string,
+  circularMask: string,
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  circularMask: boolean,
```

File: index.d.ts [hunk 13]
```diff
@@ -0,0 +1,29 @@
+interface UrlBuilderOptions {
+  allowUpscale: boolean,
+  backgroundColor: string,
+  borderRadius: string,
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  borderRadius: number,
```

File: index.d.ts [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+  borderWidth: string,
```
Comment (votemike) [inline index.d.ts]:
```suggestion
  borderWidth: number,
```

**Review by votemike — COMMENTED** [review]
Thanks.
Added some suggestions.
Perhaps looking at the tests will help?

**Comment by votemike** [comment]
Perhaps also look at the tests for valid combinations. And add the valid combinations as part of the types.
Up to you.

File: package.json [hunk 15]
```diff
@@ -4,6 +4,7 @@
   "description": "A utility to build lh3.googleusercontent URLs",
   "main": "index.js",
   "type": "module",
+  "types": "index.d.ts",
```
Comment (votemike) [inline package.json:7]: You'll probably also need to add this file to "files"

Comment (markwylde) [reply package.json:7]: I didn't see a `files` property, so I've added one now in a1d5154eff961fc5d34a36ba7c9c06c0979ff2aa.

File: index.d.ts (hunk 13 above)
Comment (markwylde) [reply index.d.ts]: Fixed in a1d5154eff961fc5d34a36ba7c9c06c0979ff2aa.

File: index.d.ts (hunk 14 above)
Comment (markwylde) [reply index.d.ts]: Fixed in a1d5154eff961fc5d34a36ba7c9c06c0979ff2aa.

File: index.d.ts (hunk 7 above)
Comment (markwylde) [reply index.d.ts:22]: Not sure how to represent this.

File: index.d.ts (hunk 3 above)
Comment (markwylde) [reply index.d.ts]: Fixed in a1d5154eff961fc5d34a36ba7c9c06c0979ff2aa.

File: index.d.ts (hunk 5 above)
Comment (markwylde) [reply index.d.ts]: Fixed in a1d5154eff961fc5d34a36ba7c9c06c0979ff2aa.

File: index.d.ts (hunk 4 above)
Comment (markwylde) [reply index.d.ts]: Fixed in a1d5154eff961fc5d34a36ba7c9c06c0979ff2aa.

File: index.d.ts (hunk 1 above)
Comment (markwylde) [reply index.d.ts]: Fixed in a1d5154eff961fc5d34a36ba7c9c06c0979ff2aa.

File: index.d.ts (hunk 2 above)
Comment (markwylde) [reply index.d.ts]: Not sure how to represent this.

File: package.json [hunk 16]
```diff
@@ -14,6 +15,10 @@
     "type": "git",
     "url": "git+https://github.com/votemike/lh3-googleusercontent-url-builder.git"
   },
+  "files": [
```
Comment (votemike) [inline package.json]: This already exists on line 11. Can you add your file there?

Comment (markwylde) [reply package.json]: Ahh, I hadn't merged in your changes from 4791fd62639af03a77403d94c28b3d760c465f0e.

**Review by votemike — APPROVED** [review]

**Pull request merged**